src/schema/search/SearchResolver.ts

```diff
@@ -0,0 +1,137 @@
+import { convertConnectionArgsToGravityArgs } from "lib/helpers"
+import { createPageCursors, pageToCursor } from "schema/fields/pagination"
+import { connectionFromArraySlice } from "graphql-relay"
+import { GraphQLResolveInfo, visit } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { Searchable } from "schema/searchable"
+import { SearchableItem } from "schema/SearchableItem"
+
+export class SearchResolver {
+  private args: any
+  private context: ResolverContext
+  private info: GraphQLResolveInfo
+  private cachedEntityTypesToFetch: string[] | undefined
+
+  constructor(args: any, context: any, info: any) {
+    this.args = args
+    this.context = context
+    this.info = info
+  }
+
+  fetch(searchResultItem) {
+    const { artistLoader, artworkLoader } = this.context
+    const loaderMapping = {
+      Artist: artistLoader,
+      Artwork: artworkLoader,
+    }
+
+    const loader = loaderMapping[searchResultItem.label]
+
+    if (loader) {
+      return loader(searchResultItem.id)
+    }
+  }
+
+  // Fetch the full object if the GraphQL query includes any inline fragments
+  // referencing the search result item's type (like Artist or Artwork)
+  shouldFetchEntityType(entityType: string): boolean {
+    if (this.cachedEntityTypesToFetch) {
```

Comment:
Tiny: If you invert this check and execute the visitor when it passes, then you only need to perform the below `includes(…)` check only at the end of the method.

Reply:
Awesome. Nice opportunity to simply the function exit :+1: